### ISSUE TITLE
Question: Optimizing Custom tile source with pixel coordinates. 

### ISSUE THREAD

**Issue by linker2424** [openseadragon/openseadragon#1308]
So, I find the custom tile source and having to have a full pyramid of images ready to go not very optimized. I was wondering if there’s any way if OSD can return the exact pixel coordinates of a mouse drag, or a zoom in or out ? Let me rephrase what I’m asking. Please keep in mind I’m new to OSD.  I don’t want to create the pyramid of tiles, I want to create the tiles base off the users panning or zooming action. So if the user is looking at a 1920 x 1080 image in a 910 x 540 viewport (scale is at 2.0) also the user is looking at exactly the center of the image in the 910 x 540 viewport, is there any way to have OSD return a new tile request for the exact pixel coordinates of the users dragging action? If the user drags his mouse left to right 20 pixel, the current image would be missing a 20 (wide) x (920 height) rectangle out of his viewport (assuming the user didn’t move up or down just a little). I just want to be able to load that user created segment/dynamic tile, but I need to know the pixel coordinates of the change in the image location the user caused. 

I don’t know if this is even possible with OSD. I’m a whiteboard person, so if this makes no sense please let me know.
Thanks so much for your time, and I’m sorry if someone already asked this.

**Comment by iangilman** [comment]
Interesting question. OSD does not support that scenario, and I don't know that it would actually be more efficient... the internet sends things in chunks, so if you're too granular with your requests, you're not actually getting any benefit. Also, if you were to zoom in a little, you'd have to get a whole new screen worth of pixels (unless you were okay with chunking things zoomwise but not panwise). 

That said, it's certainly up to you and your server whether you make the tiles ahead of time or make them on the fly based on the requests from OSD.

**Comment by linker2424** [comment]
Thank you very much for your answer. So maybe I don’t understand OSD very well yet….. Your response is leading me a bunch more questions, so I’m going to create a new question and link it here: https://github.com/openseadragon/openseadragon/issues/1311

I asked this dumb question in my other question  when I'm done with the question I'm I suppose to close it?

**Comment by iangilman** [comment]
Cool, and yes please, close the issue :)
